src/Trixi.jl

```diff
@@ -20,29 +20,39 @@ using Printf: @printf, @sprintf, println
 using Profile: clear_malloc_data
 using Random: seed!
 
-using EllipsisNotation
+using DiffEqBase: ODEProblem, get_du, u_modified!, set_proposed_dt!
+using DiffEqCallbacks: CallbackSet, DiscreteCallback
+using EllipsisNotation # ..
 using HDF5: h5open, attrs
 using StaticArrays: @MVector, @SVector, MVector, MMatrix, MArray, SVector, SMatrix, SArray
-using TimerOutputs: @notimeit, @timeit, TimerOutput, print_timer, reset_timer!
+using TimerOutputs: @notimeit, @timeit, @timeit_debug, TimerOutput, print_timer, reset_timer!
 using UnPack: @unpack
 
 # Tullio.jl makes use of LoopVectorization.jl via Requires.jl.
-# Hence, we need `using LoopVectorization` after loading Tullio and before using @tullio.
+# Hence, we need `using LoopVectorization` after loading Tullio and before using `@tullio`.
 using Tullio: @tullio
 using LoopVectorization
 
+
 # Use a central dictionary for global settings
 const globals = Dict{Symbol, Any}()
 export globals
 
+# Basic abstract types creating the hierarchy
+abstract type AbstractEquations{NDIMS, NVARS} end
+
+
+
```

Comment:
Good question. While working on this PR I thought about where we should define the entry points of our type hierarchy. I'll move the definition to `equations.jl` for now, but we should think about whether it might be nice to have the basic abstract types defined somewhere at the beginning to get an overview of them.